server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DLPRulesRoutesTest.java

```diff
@@ -0,0 +1,410 @@
+                .body("type", is("InvalidArgument"))
+                .body("message", is("The domain list does not contain: strange.com"));
+        }
+
+        @Test
+        void deleteShouldReturnBadRequestWhenDomainIsNotValid() {
+            when()
+                .delete("dr@strange.com")
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("Invalid request for domain dr@strange.com"));
+        }
+    }
+
+    @Nested
+    class DefineList {
+
+        @Test
+        void getShouldReturnOK() {
+            dlpStore.store(SENDER_DOMAIN, CONFIGURATION_ITEMS);
+
+            when()
+                .get(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON);
+        }
+
+        @Test
+        void getShouldReturnABody() {
+            dlpStore.store(SENDER_DOMAIN, CONFIGURATION_ITEMS);
+
+            String body = when()
+                .get(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body()
+                .asString();
+
+            assertThatJson(body).isEqualTo(
+                "[" +
+                "  {" +
+                "    \"id\": \"1\"," +
+                "    \"expression\": \"james.org\"," +
+                "    \"explanation\": \"explanation 1\"," +
+                "    \"targetsSender\": true," +
+                "    \"targetsRecipients\": true," +
+                "    \"targetsContent\": true" +
+                "  }," +
+                "  {" +
+                "    \"id\": \"2\"," +
+                "    \"expression\": \"james.org\"," +
+                "    \"explanation\": null," +
+                "    \"targetsSender\": true," +
+                "    \"targetsRecipients\": false," +
+                "    \"targetsContent\": false" +
+                "  }" +
+                "]");
+        }
+
+        @Test
+        void getShouldReturnAnEmptyBodyWhenDLPStoreIsEmpty() {
+            dlpStore.store(SENDER_DOMAIN, ImmutableList.of());
+
+            String body = when()
+                .get(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body()
+                .asString();
+
+            assertThatJson(body).isEqualTo("[]");
+        }
+
+        @Test
+        void getShouldReturnNotFoundWhenDomainNotInList() {
+            when()
+                .get("strange.com")
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(404))
```

Comment:
Use the HttpStatus constant for this...

Here and everywhere in this test class...